server/server.go

```diff
@@ -91,7 +91,7 @@ func (h *serverAuthProxyHandler) handler(w http.ResponseWriter, r *http.Request)
 	case http.StatusOK:
 		h.trueHandler.ServeHTTP(w, r)
 	case http.StatusUnauthorized:
-		w.Header().Set("WWW-Authenticate", "Basic realm=\"Kiali\"")
+		w.Header().Set("WWW-Authenticate", "xBasic realm=\"Kiali\"")
```

Comment:
_xBasic_ is not a standard type of authentication. If we want the backend to comply with HTTP/RFC standards, we shouldn't use it.

By using a "custom" authentication type, the browser no longer knows how to handle the authentication. This has the same effect as removing the header. So, I would remove the header instead of using _xBasic_.

Reply:
but, either use would mean that we wouldn't be able to use the api directly from the browser?! (e.g. the browser won't know that we need to provide user/pass) is that something that we want to remove?

Maybe we can instead pass a flag to turn off the passing of the WWW-Authenticate header for the login screen?

Reply:
btw, it seems that custom types are OK, but we must always provide a WWW-Authenticate header to comply the standard.
https://tools.ietf.org/html/rfc7235#section-4.1

Reply:
Mmm I am gonna check if we can avoid this and keep www-authenticate

Reply:
Oh! So my comment is somewhat wrong! :)

The authentication is, then, a little weird. Because the header will state that the server does not accept "Basic" authentication.... But it indeed accepts it, duh!

Anyway, just choose whatever you think is best.

Reply:
Or perhaps use the parameter to return something else than a 401 response? and a response that is Javascript friendly ?

Reply:
Yes, I'm only making suggestion. Choose whatever you think is best.
If you think the PR as is now it the best solution. I'm OK :)

Reply:
I think what you do already is fine, but maybe we want to pass a param (maybe a header) `X-Auth-Type-Kiali-UI=1` to signal that we don't want to use the Basic and use xBasic (or w/e).

I'm not sure how important is for devs to be able to consume the services directly from the browser, if no one raises a hand i suppose is OK to leave this PR as is